From 03b8c8b52f056cc4433cd445f4b1b8a94b7340c2 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Mon, 11 Mar 2013 14:10:40 +0100
Subject: [PATCH] configs: Update zynq defconfig to reflect changes

- enable SMC
- gpio changes because of dynamic irq allocation
- enable xilinx dma drivers + test modules

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 arch/arm/configs/xilinx_zynq_defconfig |   33 +++++++++++++++++++++++++++----
 1 files changed, 28 insertions(+), 5 deletions(-)

diff --git a/arch/arm/configs/xilinx_zynq_defconfig b/arch/arm/configs/xilinx_zynq_defconfig
index 5e55718..74ab804 100644
--- a/arch/arm/configs/xilinx_zynq_defconfig
+++ b/arch/arm/configs/xilinx_zynq_defconfig
@@ -286,6 +286,7 @@ CONFIG_ARCH_ZYNQ=y
 #
 CONFIG_XILINX_L1_PREFETCH=y
 CONFIG_XILINX_L2_PREFETCH=y
+CONFIG_DMA_ENGINE=y
 # CONFIG_XILINX_TEST is not set
 CONFIG_XILINX_AXIPCIE=y
 
@@ -380,7 +381,7 @@ CONFIG_PAGE_OFFSET=0xC0000000
 CONFIG_NR_CPUS=4
 CONFIG_HOTPLUG_CPU=y
 CONFIG_LOCAL_TIMERS=y
-CONFIG_ARCH_NR_GPIO=0
+CONFIG_ARCH_NR_GPIO=1024
 # CONFIG_PREEMPT_NONE is not set
 # CONFIG_PREEMPT_VOLUNTARY is not set
 CONFIG_PREEMPT=y
@@ -1349,7 +1350,6 @@ CONFIG_SPI_XILINX_PS_SPI=y
 #
 # Enable Device Drivers -> PPS to see the PTP clock options.
 #
-CONFIG_ARCH_HAVE_CUSTOM_GPIO_H=y
 CONFIG_ARCH_REQUIRE_GPIOLIB=y
 CONFIG_GPIOLIB=y
 CONFIG_OF_GPIO=y
@@ -2059,11 +2059,33 @@ CONFIG_RTC_DRV_PCF8563=y
 #
 # CONFIG_RTC_DRV_PL030 is not set
 # CONFIG_RTC_DRV_PL031 is not set
-# CONFIG_DMADEVICES is not set
+CONFIG_DMADEVICES=y
+# CONFIG_DMADEVICES_DEBUG is not set
+
+#
+# DMA Devices
+#
+CONFIG_XILINX_DMA_ENGINES=y
+CONFIG_XILINX_AXIDMA=y
+CONFIG_XILINX_DMATEST=y
+CONFIG_XILINX_AXIVDMA=y
+CONFIG_XILINX_VDMATEST=y
+CONFIG_XILINX_AXICDMA=y
+CONFIG_XILINX_CDMATEST=y
+# CONFIG_AMBA_PL08X is not set
+# CONFIG_DW_DMAC is not set
+# CONFIG_TIMB_DMA is not set
+# CONFIG_PL330_DMA is not set
+
+#
+# DMA Clients
+#
+# CONFIG_NET_DMA is not set
+# CONFIG_ASYNC_TX_DMA is not set
+# CONFIG_DMATEST is not set
 # CONFIG_AUXDISPLAY is not set
 CONFIG_UIO=y
 # CONFIG_UIO_CIF is not set
-CONFIG_UIO_PDRV=y
 CONFIG_UIO_PDRV_GENIRQ=y
 # CONFIG_UIO_AEC is not set
 # CONFIG_UIO_SERCOS3 is not set
@@ -2115,7 +2137,8 @@ CONFIG_RPMSG=m
 # CONFIG_VIRT_DRIVERS is not set
 # CONFIG_PM_DEVFREQ is not set
 # CONFIG_EXTCON is not set
-# CONFIG_MEMORY is not set
+CONFIG_MEMORY=y
+CONFIG_ZYNQ_SMC=y
 # CONFIG_IIO is not set
 # CONFIG_VME_BUS is not set
 # CONFIG_PWM is not set
-- 
1.7.1

